lib/theme_check/shopify_liquid/source_manager.rb

```diff
@@ -9,31 +15,68 @@ module SourceManager
 
       def download_or_refresh_files
         if has_required_files?
-          # TODO: https://github.com/Shopify/theme-check/issues/651
-          #
-          # Refresh files if they exist locally
+          refresh
         else
           download
         end
       end
 
-      def download
-        create_destination unless destination_exist?
+      def download(destination = documentation_directory)
+        create_destination(destination) unless destination_exist?(destination)
 
         required_file_names.each do |file_name|
-          download_file(local_path(file_name), remote_path(file_name))
+          download_file(local_path(file_name, destination), remote_path(file_name))
         end
       end
 
-      def local_path(file_name)
-        documentation_directory + "#{file_name}.json"
+      def refresh
+        return unless refresh_needed?
```

Comment:
I've noticed some delay in the first code completion. We might avoid that by moving this check into the thread below:

```ruby
def refresh
  @refresh_thread = Thread.new do
    if refresh_needed?
      tmp_dir = Dir.mktmpdir

      begin
        download(Pathname.new(tmp_dir))
        FileUtils.cp_r("#{tmp_dir}/.", documentation_directory)

        mark_all_indexes_outdated
      ensure
        FileUtils.remove_entry(tmp_dir)
      end
    end
  end
end
```

It's a bit uglier :( but I believe it worths as it unblocks the language server :)

Reply:
i suspect ill have to do something [like this](https://github.com/Shopify/theme-check/blob/main/lib/theme_check/remote_asset_file.rb#L34-L37) where i use `Net::HTTP.get(uri)` directly